nucliadb_node/nucliadb_node/indexer.py

```diff
@@ -0,0 +1,184 @@
+            await self.set_resource(pb)
+        elif pb.typemessage == TypeMessage.DELETION:
+            await self.delete_resource(pb)
+        else:
+            logger.warning(
+                f"Unknown type message {pb.typemessage}",
+                extra={
+                    "seqid": work.seqid,
+                    "shard": work.index_message.shard,
+                },
+            )
+
+        await signals.successful_indexing.dispatch(
+            SuccessfulIndexingPayload(seqid=work.seqid, index_message=pb)
+        )
+        logger.info(
+            "Message indexing finished",
+            extra={
+                "seqid": work.seqid,
+                "shard": pb.shard,
+                "storage_key": pb.storage_key,
+                "time": time.time() - start,
+            },
+        )
+        return True
+
+    async def set_resource(self, pb: IndexMessage) -> Optional[OpStatus]:
+        brain = await self.storage.get_indexing(pb)
+        shard_id = pb.shard
+        rid = brain.resource.uuid
+        brain.shard_id = brain.resource.shard_id = shard_id
+
+        logger.info(f"Adding {rid} at {shard_id} otx:{pb.txid}")
+        try:
+            status = await self.writer.set_resource(brain)
+
+        except AioRpcError as grpc_error:
+            if grpc_error.code() == StatusCode.NOT_FOUND:
+                logger.error(f"Shard does not exist {pb.shard}")
+            else:
+                event_id = errors.capture_exception(grpc_error)
+                logger.error(
+                    "An error ocurred on indexer worker while setting a resource. "
+                    f"Check sentry for more details. Event id: {event_id}"
+                )
+                if brain.HasField("metadata"):
+                    # Hard fail if we have the correct data
+                    raise grpc_error
+            return None
+
+        else:
+            logger.info(f"...done (Added {rid} at {shard_id} otx:{pb.txid})")
+            return status
+
+    async def delete_resource(self, pb: IndexMessage) -> Optional[OpStatus]:
+        shard_id = pb.shard
+        rid = pb.resource
+        resource = ResourceID(uuid=rid, shard_id=shard_id)
+
+        logger.info(f"Deleting {rid} in {shard_id} otx:{pb.txid}")
+        try:
+            status = await self.writer.delete_resource(resource)
+
+        except AioRpcError as grpc_error:
+            if grpc_error.code() == StatusCode.NOT_FOUND:
+                logger.error(f"Shard does not exist {pb.shard}")
+            else:
+                event_id = errors.capture_exception(grpc_error)
+                logger.error(
+                    "An error ocurred on indexer worker while deleting a resource. "
+                    f"Check sentry for more details. Event id: {event_id}"
+                )
+            return None
+
+        else:
+            logger.info(f"...done (Deleted {rid} in {shard_id} otx:{pb.txid})")
+            return status
```

Comment:
I think that the status is no longer used, so you can omit returning anything from the method (and the same for the set_resource)